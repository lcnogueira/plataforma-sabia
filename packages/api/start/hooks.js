const { hooks } = require('@adonisjs/ignitor');

const AlgoliaFakeProvider = require('../test/utils/AlgoliaFakeService');

hooks.before.providersRegistered(() => {
	if (process.env.NODE_ENV === 'testing') {
		/*
		|--------------------------------------------------------------------------
		| Add fakes
		|--------------------------------------------------------------------------
		|
		| Fake provider services before instantiation
		|
		*/
		const { ioc } = use('@adonisjs/fold');

		ioc.singletonFake('App/Services/AlgoliaSearch', () => {
			return AlgoliaFakeProvider;
		});
	}
});
