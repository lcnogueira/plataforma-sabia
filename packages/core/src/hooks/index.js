export { default as useTheme } from './useTheme';
export { default as useAuth } from './useAuth';
export { default as useTechnology } from './useTechnology';
export { default as useVisibleComponent } from './useVisibleComponent';
