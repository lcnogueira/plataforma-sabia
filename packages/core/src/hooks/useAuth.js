import { useContext } from 'react';
import { UserContext } from '../providers/UserProvider';

function useAuth() {
	return useContext(UserContext);
}

export default useAuth;
