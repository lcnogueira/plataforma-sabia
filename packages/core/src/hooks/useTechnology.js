import { useContext } from 'react';
import { TechnologyContext } from '../providers/TechnologyProvider';

function useTechnology() {
	return useContext(TechnologyContext);
}

export default useTechnology;
