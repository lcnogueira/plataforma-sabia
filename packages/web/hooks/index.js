// eslint-disable-next-line import/prefer-default-export
export { default as useModal } from './useModal';
