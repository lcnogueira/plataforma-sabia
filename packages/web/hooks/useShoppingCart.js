import { useContext } from 'react';
import ShoppingCartContext from '../components/ShoppingCart/ShoppingCartContext';

function useShoppingCart() {
	const context = useContext(ShoppingCartContext);
	if (context === undefined) {
		throw new Error('useShoppingCart must be used within a ShoppingCartProvider');
	}
	return context;
}

export default useShoppingCart;
