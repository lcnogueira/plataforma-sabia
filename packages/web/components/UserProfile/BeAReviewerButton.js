import React from 'react';
import styled, { css } from 'styled-components';
import useSWR from 'swr';
import { useAuth, useModal } from '../../hooks';
import { getReviewerUser } from '../../services/user';

const BeAReviewerButton = () => {
	const { user } = useAuth();
	const { openModal } = useModal();

	const { data: { data: currentReviewer = {} } = {}, isValidating } = useSWR(
		'get-current-reviewer',
		() => getReviewerUser(),
		{
			revalidateOnFocus: false,
		},
	);

	/*
	 * 1. Opens modal with request sent message if reviewer status equals pending
	 * 2. Opens user data pending modal if user hasnt completed registration
	 * 3. Opens modal to choose categories if previous conditions hasnt match
	 */
	const handleClick = () => {
		if (currentReviewer.status === 'pending') {
			return openModal('requestToBeReviewerSent');
		}

		if (!user.registration_completed) return openModal('pendingUserData');

		return openModal('beAReviewer');
	};

	return (
		<Button onClick={handleClick} disabled={isValidating}>
			Seja um curador
		</Button>
	);
};

const Button = styled.button`
	${({ theme: { colors } }) => css`
		background: ${colors.secondary};
		border: none;
		outline: none;
		width: 100%;
		padding: 1.6rem 3.2rem;
		margin-top: 1.2rem;
		margin-bottom: 2rem;

		color: ${colors.white};
		text-transform: uppercase;
		line-height: 2.4rem;
		font-weight: bold;

		:hover:not(:disabled),
		:focus:not(:disabled) {
			background: ${colors.darkGreen};
		}

		:disabled {
			cursor: not-allowed;
			opacity: 0.5;
		}
	`}
`;

export default BeAReviewerButton;
