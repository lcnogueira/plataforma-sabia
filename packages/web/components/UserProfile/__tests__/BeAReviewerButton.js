import React from 'react';
import { render, screen, fireEvent } from 'test-utils';

import * as useAuth from '../../../hooks/useAuth';
import BeAReviewerButton from '../BeAReviewerButton';

describe('<BeAReviewerButton />', () => {
	it('should render correctly', () => {
		jest.spyOn(useAuth, 'default').mockReturnValue({
			user: {},
		});
		const { container } = render(<BeAReviewerButton />);

		expect(screen.getByRole('button', { name: /seja um curador/i })).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('should render pending data modal when user hasnt completed registration', () => {
		jest.spyOn(useAuth, 'default').mockReturnValue({
			user: {
				registration_completed: false,
			},
		});
		render(<BeAReviewerButton />);

		fireEvent.click(screen.getByRole('button', { name: /seja um curador/i }));

		expect(screen.getByText(/você possui dados pendentes/i)).toBeInTheDocument();
		expect(screen.getByRole('button', { name: /ir para meu perfil/i })).toBeInTheDocument();
	});

	it('should render modal to choose categories when user has completed registration', () => {
		jest.spyOn(useAuth, 'default').mockReturnValue({
			user: {
				registration_completed: true,
			},
		});
		render(<BeAReviewerButton />);

		fireEvent.click(screen.getByRole('button', { name: /seja um curador/i }));

		expect(screen.getByText(/que bom poder contar contigo/i)).toBeInTheDocument();
		expect(screen.getByRole('button', { name: /enviar solicitação/i })).toBeInTheDocument();
	});
});
