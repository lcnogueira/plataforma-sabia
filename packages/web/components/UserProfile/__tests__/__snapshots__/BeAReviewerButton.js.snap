// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BeAReviewerButton /> should render correctly 1`] = `
.c0 {
  background: hsl(166,100%,34%);
  border: none;
  outline: none;
  width: 100%;
  padding: 1.6rem 3.2rem;
  margin-top: 1.2rem;
  margin-bottom: 2rem;
  color: hsla(0,0%,100%);
  text-transform: uppercase;
  line-height: 2.4rem;
  font-weight: bold;
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  background: hsla(146,64%,36%);
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

<div>
  <button
    class="c0"
    disabled=""
  >
    Seja um curador
  </button>
</div>
`;
